playwright-tests/pages/dashboards/BaseDashboard.page.ts

```diff
@@ -0,0 +1,143 @@
+
+export class BaseDashboard extends CommonPage {
+  constructor(page: Page) {
+    super(page);
+  }
+      
+    
+
+  private baseDashboardElements = {
+    ...super.getElements(),
+    collapsedPanel: this.page.locator('//*[contains(@class, "dashboard-row--collapsed")]'),
+    panelContent: this.page.locator('//*[contains(@class, "panel-content")]'),
+    panel: this.page.locator('//div[contains(@class, "react-grid-item")]'),
+    panelTitle: this.page.locator('//div[@class="panel-title"]//h2'),
+    getPanelByName: (name: string, panelId: number) =>  this.page.locator(`//*[text()="${name}"]//ancestor::div[contains(@class, "react-grid-item")  and @data-panelId="${panelId}"]`),
+  };
+
+  private baseDashboardFields = {
+    ...super.getFields(),
+  };
+
+  private baseDashboardLabels = {
+    ...super.getLabels(),
+  };
+
+  private baseDashboardButtons = {
+    ...super.getButtons(),
+    qan: this.page.locator('//*[@data-testid="data-testid Dashboard link"]//span[text()="Query Analytics"]'),
+  };
+
+  private baseDashboardMessages = {
+    ...super.getMessages(),
+  };
+
+  private baseDashboardLinks = {
+    ...super.getLinks(),
+  };
+
+  protected getBaseDashboardElements() {
+    return this.baseDashboardElements;
+  }
+
+  protected getBaseDashboardFields() {
+    return this.baseDashboardFields;
+  }
+
+  protected getBaseDashboardLabels() {
+    return this.baseDashboardLabels;
+  }
+
+  protected getBaseDashboardButtons() {
+    return this.baseDashboardButtons;
+  }
+
+  protected getBaseDashboardMessages() {
+    return this.baseDashboardMessages;
+  }
+
+  protected getBaseDashboardLinks() {
+    return this.baseDashboardLinks;
+  }
+
+  openAllPanels = async () => {
+    try {
+      await this.baseDashboardElements.collapsedPanel.waitFor({ state: 'visible' });
+    } catch (e) {}
+
+    const collapsedPanels = await this.baseDashboardElements.collapsedPanel.elementHandles()
+    for await (const [index, panel] of collapsedPanels.entries()) {
+      await panel.click()
+    }
+    await this.page.keyboard.press('PageDown');
+    await this.page.waitForTimeout(1000);
+    await this.page.keyboard.press('PageDown');
+    await this.page.waitForTimeout(1000);
+    await this.page.keyboard.press('PageDown');
+    await this.page.waitForTimeout(1000);
+    await this.page.keyboard.press('PageDown');
+    await this.page.waitForTimeout(1000);
+    await this.page.keyboard.press('PageUp');
+    await this.page.waitForTimeout(1000);
+    await this.page.keyboard.press('PageUp');
+    await this.page.waitForTimeout(1000);
+  }
+
+  waitForPanelToHaveData = async (panelHeader: string, panelId: number, timeout: Duration = Duration.OneMinute) => {
+    await this.openAllPanels();
+    await this.baseDashboardElements.getPanelByName(panelHeader, panelId).scrollIntoViewIfNeeded();
+    await expect(this.baseDashboardElements.getPanelByName(panelHeader, panelId)).not.toContainText('N/A' && 'No data', { ignoreCase: true, timeout });
```

Comment:
Could this give false positive? Because the text will never be both 'N/A' && 'No data'?

Reply:
I have added separate check for N/A, no data, and permission issue.